### DISCUSSION TITLE
Subset IP used in actions

### DISCUSSION THREAD

**Discussion by filipemsc** [community/community#29595] [Actions] [answered]
I want to run a code on Github Actions that access an FTP that is not mine and blocks IP addresses outside Brazil. Can I restrict the Github Actions to only use Brazil's IP addresses? Is there another way of solving this problem?

**Comment by luckylittle** [comment, marked as answer]
@filipemsc To get a list of IP address ranges that GitHub Actions uses for GitHub-hosted runners, you can use the GitHub REST API:

```bash
curl -H "Accept: application/vnd.github+json" -H "Authorization: token <OMITTED>" https://api.github.com/meta | jq .actions
```

Today (2022-08-16), there are 2370 IP addresses in that list. But that list changes every week, so whitelisting might be hard.

In general, Windows and Ubuntu runners are hosted in Azure and subsequently have the same IP address ranges as the Azure datacenters. macOS runners are hosted in GitHub's own macOS cloud.

Generally, GitHub hosts runners are in the following Microsoft Azure regions:
* East US (eastus)
* East US 2 (eastus2)
* West US 2 (westus2)
* Central US (centralus)
* South Central US (southcentralus)

Unfortunately you aren’t able to choose which runner locations are used though, so you might need to use [Self Hosted Runners](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners) and host them wherever you need (Brazil).

_Note: Please accept the answer if it answers your question, thanks_
